### PR TITLE
feat: add valibot

### LINKED ISSUE #1104: Add `valibot`
It would be awesome to see its metrics 👀 

@fabian-hiller https://github.com/fabian-hiller/valibot

https://www.builder.io/blog/introducing-valibot

### PR DESCRIPTION
**Pull request by naruaway**
We might want to wait for https://github.com/fabian-hiller/valibot/pull/46 to be resolved since I expect that could improve the performance a lot

## (Benchmark on my MacBook Air M2)

I ran `npm run start run valibot`:

Against `valibot@0.8.0`:
```
Removing previous results
Executing "valibot"
Loading "valibot"
Running "parseSafe" suite...
Progress: 100%

  valibot:
    303 700 ops/s, ±2.28%   | fastest

Finished 1 case!
Running "parseStrict" suite...
Progress: 100%

  valibot:
    338 906 ops/s, ±2.54%   | fastest

Finished 1 case!
```

Against https://github.com/fabian-hiller/valibot/pull/46:
```
Removing previous results
Executing "valibot"
Loading "valibot"
Running "parseSafe" suite...
Progress: 100%

  valibot:
    659 443 ops/s, ±1.68%   | fastest

Finished 1 case!
Running "parseStrict" suite...
Progress: 100%

  valibot:
    691 575 ops/s, ±1.99%   | fastest

Finished 1 case!
```

Closes #1104

### CODE REVIEW

**Comment by moltar** [comment]
@naruaway Thank you for your contribution!! 🎉 

It looks like there is a bit of an issue with some lints.

Could you please run `npm run lint:fix`?

---

> We might want to wait for https://github.com/fabian-hiller/valibot/pull/46 to be resolved since I expect that could improve the performance a lot

Up to you! But Renovate will pick it up as soon as it's released and we can merge the update too. Unless there'll be some breaking changes, of course.

**Comment by naruaway** [comment]
Thanks! I fixed the lint 🙇 
> But Renovate will pick it up as soon as it's released

Nice, then probably fine to merge without waiting any change in Valibot

**Comment by tonivj5** [comment]
Amazing! it closes https://github.com/moltar/typescript-runtime-type-benchmarks/issues/1104

File: README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@
 * [Vality](https://github.com/jeengbe/vality)
 * [yup](https://github.com/jquense/yup)
 * [zod](https://github.com/vriad/zod)
+* [valibot](https://github.com/fabian-hiller/valibot)
```
Comment (moltar) [inline README.md:51]: Oops, sorry, just noticed this. The list is in alphabetical order. Could you please sort it? Thanks!!

**Review by moltar — CHANGES_REQUESTED** [review]

**Comment by moltar** [comment]
Build passes otherwise! 🎉

File: README.md (hunk 1 above)
Comment (naruaway) [reply README.md]: Sure! Fixed now 👍

**Comment by naruaway** [comment]
I realized that this change was not enough for valibot to appear in the benchmark. I created https://github.com/moltar/typescript-runtime-type-benchmarks/pull/1116 🙏

**Comment by fabian-hiller** [comment]
Thank you!

**Pull request merged**